R/tm_g_response.R

```diff
@@ -234,7 +234,44 @@ srv_g_response <- function(id,
   moduleServer(id, function(input, output, session) {
     data_extract <- list(response = response, x = x, row_facet = row_facet, col_facet = col_facet)
 
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+    rule_diff <- function(other) {
+      function(value) {
+        othervalue <- selector_list()[[other]]()$select
+        if (identical(value, othervalue))
+          "Row and column facetting variables must be different."
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = data_extract,
+      datasets = data,
+      select_validation_rule = list(
+        response = shinyvalidate::sv_required("Please define a column for the response variable"),
+        x = shinyvalidate::sv_required("Please define a column for X variable"),
+        row_facet = shinyvalidate::compose_rules(
+          shinyvalidate::sv_optional(),
+          ~ if (length(.) > 1) "There must be 1 or no row facetting variable.",
+          crule(rule_diff("col_facet"), !is.null(input$row_facet))
+        ),
+        col_facet = shinyvalidate::compose_rules(
+          shinyvalidate::sv_optional(),
+          ~ if (length(.) > 1) "There must be 1 or no column facetting variable.",
+          crule(rule_diff("row_facet"), !is.null(input$col_facet))
```

Comment:
has crule been removed though? It's here twice...

Reply:
I missed it in the roll back. Fixed.